### ISSUE TITLE
[WEBFORM-1001] : Thumbnails are taking too much time in loading for large files

### ISSUE THREAD

**Issue by usmanazizgroupdocs** [groupdocs-viewer/GroupDocs.Viewer-for-.NET#36]
This issue was originally raised via forum thread: [http://groupdocs.com/Community/forums/thread/12584/how-to-process-file-from-stream-without-filename.aspx](http://groupdocs.com/Community/forums/thread/12584/how-to-process-file-from-stream-without-filename.aspx)

Thumbnails are taking a considerable time in loading for large files in WebForms Front End.

**Comment by PuneetRajak** [comment]
Hi 
Please let us know if this issue is fixed or when you are expecting to deliver fixes of this and related performance concerns.
Puneet Rajak

**Comment by groupdocs-viewer** [comment]
Hi Puneet,

Thank you for your query,

In fact, the thumbnails generation is dependent on the back-end API. Currently, the API takes a lot of time even to get pre-load page count. Hence, this issue would be improved automatically, as the performance issue like [your this forum thread](http://groupdocs.com/Community/forums/thread/13555/preload-page-count-are-still-slow-as-compare-to-v2.x-version-of-viewer.aspx) would be resolved.

Many Thanks.

**Comment by PuneetRajak** [comment]
Hi
is there any update on this.
What we analyse from observing the Cache folder is that thumbnail only load when the complete caching of the document take place(including the last page of document).
And the complete caching of the document takes a big time due to slow caching performance. 
So in conclusion, the loading of thumbnail is also affected by viewer performance.
If we set useCache = false, then also it take same amount of time
so in conclusion it more look like performance is hampering when opening large document.
Thanks,

**Comment by muhammadumargroupdocs** [comment]
Hi Puneet,

Thank you for your work around.

I agree with your observations. The issue is really appearing in the case of large size files and these performance based issues are actually core level issues of the back-end API. Hence, these issues are already logged in the issue tracking system for the back-end API. Unfortunately, no updates are available for these back-end API issues yet. We'll get back to you once we found any update regarding this particular issue.
 
Thanks.

**Comment by PuneetRajak** [comment]
Hi
I can see there is a new release note
http://www.groupdocs.com/docs/display/viewernet/GroupDocs.Viewer+For+.NET+16.10+Release+Notes
please let me know would that v16.10 has fixes for our issues?

Thanks
Puneet Rajak

**Comment by muhammadumargroupdocs** [comment]
Hi Puneet,

Thank you for your note.

Unfortunately, performance based issues couldn't be fixed in the latest monthly released. No updates could be found from the product team. However, We'll notify you whenever these particular issues would be resolved.

Thank you

**Comment by PuneetRajak** [comment]
Hi
does the new version v16.11 has fixes for our raised issue !

Thanks
Puneet R

**Comment by usmanazizgroupdocs** [comment]
Hi Puneet,

Thanks for writing to us.

We apologize that **GroupDocs.Viewer for .NET 16.11.0** does not contain the fix for performance specific issues. However, we would like you to upgrade your application to the latest version of the API to get a better experience. As we get any further updates related to the performance related issues, we will keep you updated.

Warm Regards

**Comment by usmanazizgroupdocs** [comment]
We have discontinued the support of old front ends including _ASP.NET_MVC_Front_End_ and _ASP.NET_WebForm_Front_End_. Therefore, we recommend you to migrate to our latest front ends (listed below) and share your feedback with us. 

- [GroupDocs.Viewer-for-.NET-MVC-App](https://github.com/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC-App)
- [GroupDocs.Viewer-for-.NET-WebForms-App](https://github.com/groupdocs-viewer/GroupDocs.Viewer-for-.NET-WebForms-App)

Regards
